CMakeLists.txt flow control statement nesting issue

Trying to build `chai` using spack -- under https://github.com/spack/spack/pull/25321 -- and getting the following error (that also appeared when building `care` and `glvis`):

```
...
==> [2021-08-12-00:04:42.293990] chai: Building chai-develop-wzzsm5uvishgiebaczrwbzakg646fb3m [CMakePackage]
==> [2021-08-12-00:04:42.310775] chai: Executing phase: 'cmake'
==> [2021-08-12-00:04:42.319438] 'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/home/software/radiuss/[padded-to-512-chars]/morepadding/linux-ubuntu18.04-x86_64/gcc-7.5.0/chai-develop-wzzsm5uvishgiebaczrwbzakg646fb3m' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/home/software/radiuss/[padded-to-512-chars]/morepadding/linux-ubuntu18.04-x86_64/gcc-7.5.0/chai-develop-wzzsm5uvishgiebaczrwbzakg646fb3m/lib;/home/software/radiuss/[padded-to-610-chars]/morepadding/linux-ubuntu18.04-x86_64/gcc-7.5.0/chai-develop-wzzsm5uvishgiebaczrwbzakg646fb3m/lib64;/home/software/radiuss/[padded-to-612-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/umpire-develop-iztrmuhvzvvffh32phtiyy62u3fdvjtx/lib;/home/software/radiuss/[padded-to-600-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/camp-master-kf2ofmqh54quzf4bysluyumymjrszhen/lib' '-DCMAKE_PREFIX_PATH:STRING=/home/software/radiuss/[padded-to-512-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/umpire-develop-iztrmuhvzvvffh32phtiyy62u3fdvjtx;/home/software/radiuss/[padded-to-596-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/camp-master-kf2ofmqh54quzf4bysluyumymjrszhen;/home/software/radiuss/[padded-to-593-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/blt-0.4.1-w3rx3gge3lvk5u63sh2rr2f6zfarqcrz;/home/software/radiuss/[padded-to-591-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/cmake-3.21.1-4uhxhmxa4yv6z4nzjnskjrwgmwmsduik' '-DBLT_SOURCE_DIR=/home/software/radiuss/[padded-to-512-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/blt-0.4.1-w3rx3gge3lvk5u63sh2rr2f6zfarqcrz' '-DENABLE_CUDA=OFF' '-DENABLE_HIP=OFF' '-DENABLE_PICK:BOOL=ON' '-Dumpire_DIR:PATH=/home/software/radiuss/[padded-to-512-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/umpire-develop-iztrmuhvzvvffh32phtiyy62u3fdvjtx/share/umpire/cmake' '-DENABLE_TESTS=OFF' '-DENABLE_BENCHMARKS:BOOL=OFF' '-DENABLE_EXAMPLES:BOOL=ON' '-DENABLE_BENCHMARKS=OFF' '/tmp/root/spack-stage/spack-stage-chai-develop-wzzsm5uvishgiebaczrwbzakg646fb3m/spack-src'
CMake Error at CMakeLists.txt:50 (endif):
  Flow control statements are not properly nested.


-- Configuring incomplete, errors occurred!
...
```

Adding these packages to Spack Cloud CI PR builds is being deferred until the impact of the fix is assessed.

[chai-develop-cmakelist-flow-error.txt](https://github.com/LLNL/CHAI/files/6985270/chai-develop-cmakelist-flow-error.txt)


Looking at the code, it appears the error has been around since a commit 10 months ago.  See line https://github.com/LLNL/CHAI/blob/be562768728e916774d3a156c51fb062a4329b25/CMakeLists.txt#L49.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,9 +44,8 @@ set(ENABLE_COPY_HEADERS Off CACHE BOOL "")
 set(BLT_CXX_STD c++11 CACHE STRING "")
 
 if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI")
-    message(STATUS "Setting CMAKE_CXX_EXTENSIONS to ON for PGI Compiler")
-    set( CMAKE_CXX_EXTENSIONS ON )
-  endif()
+  message(STATUS "Setting CMAKE_CXX_EXTENSIONS to ON for PGI Compiler")
+  set( CMAKE_CXX_EXTENSIONS ON )
 endif()
 
 if (ENABLE_CUDA)

